ci-operator/step-registry/rosa/sts/account-roles/create/rosa-sts-account-roles-create-commands.sh

```
@@ -52,32 +52,27 @@ if [[ "$HOSTED_CP" == "true" ]]; then
+    OPENSHIFT_VERSION=$(echo "$ORIGINAL_RELEASE_IMAGE_LATEST" | sed -E 's/.*:([0-9]+\.[0-9]+).*/\1/')
   fi
+fi
 
-  OPENSHIFT_VERSION=$(echo "${OPENSHIFT_VERSION}" | cut -d '.' -f 1,2)
-
-  # Determine cluster type switch early so the version fallback can use it
-  CLUSTER_SWITCH="--classic"
+if [[ -z "$OPENSHIFT_VERSION" ]]; then
   if [[ "$HOSTED_CP" == "true" ]]; then
-    CLUSTER_SWITCH="--hosted-cp"
+    versionList=$(rosa list versions --channel-group ${CHANNEL_GROUP} --hosted-cp -o json | jq -r '.[].raw_id')
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)